bigtable/client/rpc_backoff_policy.cc

```diff
@@ -46,12 +46,22 @@ void ExponentialBackoffPolicy::setup(grpc::ClientContext& /*unused*/) const {}
 std::chrono::milliseconds ExponentialBackoffPolicy::on_completion(
     grpc::Status const& status) {
   using namespace std::chrono;
-  // TODO(#57) - we also need to randomize the sleep period.
   auto delay = duration_cast<milliseconds>(current_delay_);
-  current_delay_ *= 2;
-  if (current_delay_ >= maximum_delay_) {
-    current_delay_ = maximum_delay_;
+  std::uniform_int_distribution<int> rng_distribution(
+      0, duration_cast<microseconds>(current_delay_).count());
+  // Randomized sleep period because it is possible that after some time all
+  // client have same sleep period if we use only exponential backoff policy.
+  auto random_delay = microseconds(rng_distribution(generator_));
+  if (random_delay <= current_delay_) {
+    std::uniform_int_distribution<int> distribution((2 * random_delay).count(),
+                                                    (3 * random_delay).count());
+    random_delay = microseconds(distribution(generator_));
   }
+  random_delay *= 2;
```

Comment:
Doesn't this still have the problem that once `current_delay_` reaches `maximum_delay_` then the function returns `maximum_delay_` from that point forward?  Or I am reading this wrong?

Reply:
Changed it to generate a one close maximum_delay.